### PR TITLE
AUTH-438: Added documentation for ZIP distribution

### PR DESCRIPTION
**Pull request by ehardon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+
+#### Upgrade example for Identity Service with Postgres DB
+
+1. Backup the old installation performing:
+
+	```bash
+	pg_dump --clean --no-owner --no-acl -h ${POSTGRES_HOST} -p ${POSTGRES_PORT}  -U ${POSTGRES_USER} ${POSTGRES_DATABASE} | grep -v -E '(DROP\ SCHEMA\ public|CREATE\ SCHEMA\ public|COMMENT\ ON\ SCHEMA\ public|DROP\ EXTENSION\ plpgsql|CREATE\ EXTENSION\ IF\ NOT\ EXISTS\ plpgsql|COMMENT\ ON\ EXTENSION\ plpgsql)' > /backup/backup.sql
+	```
+	
+2. Remove old data and stop the postgres instance.
```
Comment (cristina-sirbu) [inline README.md]:
```suggestion
2. Remove old data and stop the PostgreSQL instance.
```

File: README.md [hunk 2]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+
+6. Upgrade the adapters.
+
+Within the next sections we will go trough a simple distribution and kubernetes upgrade plus rollback.
+
+  **_NOTE:_** In depth documentation on keycloak upgrade can be found [here](https://www.keycloak.org/docs/7.0/upgrading/index.html#_upgrading).
+
+### ZIP Distribution
+
+#### Upgrade example for Identity Service with Postgres DB
```
Comment (cristina-sirbu) [inline README.md]:
```suggestion
#### Upgrade example for Identity Service with PostgreSQL database
```

File: README.md [hunk 3]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+
+	```bash
+	pg_dump --clean --no-owner --no-acl -h ${POSTGRES_HOST} -p ${POSTGRES_PORT}  -U ${POSTGRES_USER} ${POSTGRES_DATABASE} | grep -v -E '(DROP\ SCHEMA\ public|CREATE\ SCHEMA\ public|COMMENT\ ON\ SCHEMA\ public|DROP\ EXTENSION\ plpgsql|CREATE\ EXTENSION\ IF\ NOT\ EXISTS\ plpgsql|COMMENT\ ON\ EXTENSION\ plpgsql)' > /backup/backup.sql
+	```
+	
+2. Remove old data and stop the postgres instance.
+
+3. Stop the Identity Service 1.1 server.
+
+4. Open the 1.2 Identity Service distribution zip and configure accordingly to the database that will be used (for this example Postgres).
```
Comment (cristina-sirbu) [inline README.md]:
```suggestion
4. Open Identity Service 1.2 distribution zip and configure accordingly to the database that will be used (for this example PostgreSQL).
```

File: README.md [hunk 4]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+   For detailed information on how to set up the desired database this visit the official documentation of Keycloak [here](https://www.keycloak.org/docs/4.8/server_installation/#_database).
+   
+5. Start the database and restore data by executing the following command:
+
+ ```bash
+  psql -h ${POSTGRES_HOST} -p ${POSTGRES_PORT} -d ${POSTGRES_DATABASE} -U ${POSTGRES_USER} -f /backup/backup.sql
+ ``` 
+
+6. Start Identity Service 1.2.
```
Comment (cristina-sirbu) [inline README.md]: How do you start Identity Service?

File: README.md [hunk 5]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
 
+## Upgrading from Identity Service 1.1 to 1.2
+
+  **_NOTE:_** The upgrade of the Alfresco Identity Management Service requires downtime. 
+  This means that no user will be able to connect to any of the Digital Business Platform components while the upgrade or rollback is being done.
+
+### General upgrade procedure
+
+For upgrading Alfresco Identity Management Service we are mainly following the Keycloak upgrade procedure.
+We will be explaining how to do it if you are using our out of the box distribution or Kubernetes deployment.
```
Comment (rgauss) [inline README.md]: > ... using our zip distribution or ...

File: README.md [hunk 6]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+However depending on the environment you are using you should follow the next steps:
```
Comment (rgauss) [inline README.md]: > you should follow these high-level steps:

File: README.md [hunk 7]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+
+1. Prior to applying the upgrade, handle any open transactions and delete the data/tx-object-store/ transaction directory.
```
Comment (rgauss) [inline README.md]: Same [comment as on the other PR](https://github.com/Alfresco/alfresco-identity-service/pull/113#discussion_r353145678)

I suppose we'll have merge conflicts with this sections that we'll need to resolve depending on which PR gets in first.

File: README.md [hunk 8]
```diff
@@ -288,6 +289,64 @@ Once Keycloak is up and running, login to the [Management Console](http://www.ke
...
+  **_NOTE:_** In depth documentation on keycloak upgrade can be found [here](https://www.keycloak.org/docs/7.0/upgrading/index.html#_upgrading).
+
+### ZIP Distribution
+
+#### Upgrade example for Identity Service with Postgres DB
+
+1. Backup the old installation performing:
+
+	```bash
+	pg_dump --clean --no-owner --no-acl -h ${POSTGRES_HOST} -p ${POSTGRES_PORT}  -U ${POSTGRES_USER} ${POSTGRES_DATABASE} | grep -v -E '(DROP\ SCHEMA\ public|CREATE\ SCHEMA\ public|COMMENT\ ON\ SCHEMA\ public|DROP\ EXTENSION\ plpgsql|CREATE\ EXTENSION\ IF\ NOT\ EXISTS\ plpgsql|COMMENT\ ON\ EXTENSION\ plpgsql)' > /backup/backup.sql
```
Comment (rgauss) [inline README.md]: Doesn't our zip distribution use H2 by default?

Should we have a note that says the instructions assume a postrgresql has been used in a production environment?

Comment (ehardon) [reply README.md]: These instructions are under Upgrade example for Identity Service with PostgreSQL database. Is it ok? I think in the other merge request there's something written about the default H2 DB.

Comment (rgauss) [reply README.md]: Ah, yes, I guess I missed that header due to the PR rendering.

**Comment by rgauss** [comment]
This will end up being a huge section when all PRs are merged.  We should consider moving this to its own page with a link in the main README after all merges are complete.

**Pull request closed without merging**